File: 011_Container_With_Most_Water/011_2.py
class Solution:
    def maxArea(self, height):
        i, j = 0, len(height) - 1
        area = 0
        while i < j:
            area = max(area, min(height[i], height[j]) * (j - i))
            if height[i] <= height[j]:
                i += 1
            else:
                j -= 1
        return area

File: 011_Container_With_Most_Water/011_2.java
class Solution
{
    public int maxArea(int[] height)
    {
        int i = 0;
        int j = height.length - 1;
        int area = Math.min(height[i], height[j]) * j, area_2;

        while (i < j)
        {
            if (height[i] <= height[j]) i++;
            else                        j--;
            area_2 = Math.min(height[i], height[j]) * (j - i);
            if (area < area_2) area = area_2;
        }
        return area;
    }
}